tests/ext-activitylog.test.js

```diff
@@ -269,3 +270,77 @@ test('clearing logs from activitylog page', async () => {
   expect(activityLog.model.logs).toMatchObject([]);
   expect(getTableRows().length).toBe(0);
 });
+
+test('timestamp is formatted and rendered correctly', () => {
+  const logTimestamp = 1597686226302;
+
+  const logs = [
+    {
+      /* renders 1st row in table */
+      id: 'id1@test',
+      viewType: 'viewType@test',
+      type: 'type@test',
+      data: [{ test: 'test1@data' }],
+      timeStamp: logTimestamp,
+    },
+  ];
+
+  const addListener = jest.fn();
+  const sendMessage = jest.fn();
+  const connect = jest.fn();
+
+  window.browser = {
+    runtime: {
+      onMessage: { addListener },
+      sendMessage,
+      connect,
+    },
+  };
+
+  sendMessage.mockImplementation(() => {
+    return Promise.resolve({ existingLogs: [] });
+  });
+
+  document.body.innerHTML = activityLogBody;
+  const dateTimeFormatFn = jest.spyOn(ExtListen, 'dateTimeFormat');
+
+  const { activityLog } = new ActivityLog();
+
+  // To have consistant date format, we choose "en-US" formatting
+  let expectedDate;
+  dateTimeFormatFn.mockImplementation((timestamp, options) => {
+    const dateTime = new Date(timestamp);
+    const time = dateTime.toLocaleTimeString();
+
+    if (options?.timeOnly) {
+      return time;
+    }
+
+    const dateFormatOptions = {
+      month: 'short',
+      day: 'numeric',
+      year: 'numeric',
+    };
+    expectedDate = new Intl.DateTimeFormat('en-US', dateFormatOptions).format(
+      dateTime
+    );
+
+    return `${expectedDate} ${time}`;
+  });
+
+  activityLog.handleNewLogs(logs);
+
+  expect(dateTimeFormatFn).toHaveBeenCalled();
+
+  const dateTime = new Date(logTimestamp);
+  const expectedTime = dateTime.toLocaleTimeString();
```

Comment:
Hard-code the expected strings instead of generating it based on the dynamic objects.